### ISSUE TITLE
npm run win-local doesn't work

### ISSUE THREAD

**Issue by sclark39** [awslabs/aws-serverless-express#38]
This lib seems to use unix sockets for the proxy, which won't work on windows. I was able to get it to work by modifying the http request and server listen calls to use localhost + port.

**Comment by brettstack** [comment]
Can you provide example and potentially PR?

**Comment by sclark39** [comment]
The issue exists with the example project. When executing `npm run win-local`, node launches and then the process immediately exits without any output. It never successfully creates a server nor handles the sample event.

**Comment by SethDG** [comment]
Agreed over here.  Neither win-local or win-setup work for me.  Attached is my log from win-local


[2017-04-14T18_22_31_791Z-debug.log.txt](https://github.com/awslabs/aws-serverless-express/files/922699/2017-04-14T18_22_31_791Z-debug.log.txt)
[2017-04-14T18_22_31_891Z-debug.log.txt](https://github.com/awslabs/aws-serverless-express/files/922698/2017-04-14T18_22_31_891Z-debug.log.txt)

**Comment by SethDG** [comment]
And win-setup


[2017-04-14T18_17_16_070Z-debug.log.txt](https://github.com/awslabs/aws-serverless-express/files/922713/2017-04-14T18_17_16_070Z-debug.log.txt)
[2017-04-14T18_17_16_139Z-debug.log.txt](https://github.com/awslabs/aws-serverless-express/files/922712/2017-04-14T18_17_16_139Z-debug.log.txt)
[2017-04-14T18_17_16_207Z-debug.log.txt](https://github.com/awslabs/aws-serverless-express/files/922711/2017-04-14T18_17_16_207Z-debug.log.txt)

**Comment by talawahtech** [comment]
I just ran into this issue as well. Running node.js 6.10 on Windows 10 I got this error: 

`Error: listen EACCES /tmp/server0.sock`

[The offending line of code](https://github.com/awslabs/aws-serverless-express/blob/505b0958682ad6898deb6d26073ce22100242e36/index.js#L140) will only work on unix-like systems. I took a quick look at the relevant docs[1][2] and came up with a quick workaround that uses windows pipes instead of sockets if you are on windows.

```
function getSocketPath(socketPathSuffix) {
  if (/^win/.test(process.platform)) {
    const path = require('path')
    return path.join('\\\\?\\pipe', process.cwd(), `server${socketPathSuffix}`);
  }
  else {
    return `/tmp/server${socketPathSuffix}.sock`
  }
}
```


1. https://nodejs.org/api/net.html#net_server_listen_path_backlog_callback
2. https://nodejs.org/api/net.html#net_identifying_paths_for_ipc_connections

**Comment by lems3** [comment]
@talawahtech I just updated the code on my local version, it works on Windows 10. Could you or someone else make a PR and add this to the project ?

I can look into it next week if no one has time to work on this.

**Comment by brettstack** [comment]
Fixed by #100
